src/node.cc

```diff
@@ -3626,6 +3631,7 @@ static void CheckIfAllowedInEnv(const char* exe, bool is_env,
     "--pending-deprecation",
     "--no-warnings",
     "--napi-modules",
+    "--preserve-dns-order",
```

Comment:
> Need discussion for whether this should be added to --help output

Note the comments for the whitelist:
```
// Node options, sorted in `node --help` order for ease of comparison.
```

so if it's decided not to add this option to `--help` then this option should be moved into another commented block (e.g. `// Node options, not listed in 'node --help'`) before the V8 options.